docker/Dockerfile

```diff
@@ -130,7 +129,7 @@ WORKDIR /cyclus
 
 # You may add the option "--cmake-debug" to the following command
 # for further CMake debugging.
-RUN python3 install.py -j ${make_cores} --allow-milps
+RUN python3 install.py -j ${make_cores} --allow-milps --parallel
```

Comment:
Do we test that it still builds correctly without this flag?

Reply:
I added a build arg to the dockerfile and a matrix to the workflow.  We can also add a matrix for `--allow-milps` (or any other optional build flags we may want in the future), see `macos_build_test.yml`.  I didn't add it to `build_test.yml` yet because the cycamore Dockerfile doesn't support passing build flags through a `--build-arg` yet. We can add a coin_flags matrix to build_test.yml in a PR following this one (and a cycamore PR) assuming its desired